source/slang/slang-ast-dump.cpp

```diff
@@ -304,9 +304,9 @@ struct ASTDumpContext
 
     void dump(uint32_t v)
     {
-        m_writer->emit(UInt(v));
+        m_writer->emit((uint64_t)v);
     }
-    void dump(UInt v)
```

Comment:
(With WASM target, `UInt` is the same type as `uint32_t` and so the function was redefined.)

Reply:
There may be some Emscripten options (`-sWASM_BIGINT` and/or `-sMEMORY64=1`) that I believe would let us have `long` be 64 bits...

Reply:
We should probably turn that flag on.

Reply:
I experimented with these flags but for some reason they don't seem to help with these particular issues. (In fact I think these are linker flags.)

64 bit integers are emulated in JS and so there's be a perf hit involved.
https://emscripten.org/docs/porting/guidelines/portability_guidelines.html#code-that-compiles-but-might-run-slowly

`WASM_BIGINT` just skips a legalization steps where i64s become pairs of i32s in favor of emitting i64 operations and relying on special VM/browser support for i64s. I guess whatever flag makes long be 64 bits is orthogonal.

In any case I'd like to defer the decision since it seems to be working as it is right now.